python/BioSimSpace/Protocol/_free_energy.py

```diff
@@ -136,6 +143,31 @@ def __repr__(self):
                    ) % (self._lambda, self._lambda_vals, self._timestep, self._runtime,
                         self._temperature, self._pressure, self._report_interval, self._restart_interval)
 
+    def getPertType(self):
+        """Get the type of step of the multistep approach to write. Default ("standard") is onestep.
+
+           Returns
+           -------
+
+           pert_type : str
+               The perturbation type.
+        """
+        return self._pert_type
+
+    def setPertType(self, pert_type):
+        """Set the time step.
+
+           Parameters
+           ----------
+
+           pert_type : str
+               The perturbation type.
+        """
+        if type(pert_type) is str:
+            self._pert_type = pert_type
+        else:
+            raise TypeError("'pert_type' must be of type str")
```

Comment:
Could you also add the check that you perform in `_SireWrappers/_molecule.py` here, i.e.:

```python
          if type(pert_type) is not str:
              raise TypeError("'pert_type' must be of type 'str'")

          allowed_pert_types = [
                                "standard",
                                "1_discharge_soft",
                                "2_vanish_soft",
                                "3_flip",
                                "4_grow_soft",
                                "5_charge_soft"]
          if pert_type not in allowed_pert_types:
            raise ValueError("'pert_type' must be any of: "+str(allowed_pert_types), pert_type)
```

I would also add `perturbation_type = perturbation_type.lower().replace(" ", "")` to catch case and whitespace errors before checking against the list of allowed values.

It's good to catch errors as soon as they occur, rather than at the point at which the parameter is used. Still leave the check in `_molecule.py`, since that will catch the same errors if a user happens to use the `_toPertFile` method directly.

Reply:
see 2fa10bb54f3315a5d2cf7a44d16224e51d2392d4